test/jdk/com/sun/jdi/JdwpNetProps.java

```diff
@@ -60,33 +61,82 @@ public static void main(String[] args) throws Exception {
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(false)
+                    .preferIPv6Addresses("true")
+                    .run(TestResult.AttachFailed);
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(false)
+                    .preferIPv6Addresses("system")
+                    .run(systemPrefersIPv6 ? TestResult.AttachFailed : TestResult.Success);
             if (ipv6Address != null) {
-                // - only IPv4, so connection prom IPv6 should fail
+                // - only IPv4, so connection from IPv6 should fail
+                new ListenTest("localhost", ipv6Address)
+                        .preferIPv4Stack(true)
+                        .preferIPv6Addresses("true")
+                        .run(TestResult.AttachFailed);
                 new ListenTest("localhost", ipv6Address)
                         .preferIPv4Stack(true)
-                        .preferIPv6Addresses(true)
+                        .preferIPv6Addresses("system")
                         .run(TestResult.AttachFailed);
                 // - listen on IPv4
                 new ListenTest("localhost", ipv6Address)
-                        .preferIPv6Addresses(false)
+                        .preferIPv6Addresses("false")
+                        .run(TestResult.AttachFailed);
+                // - listen on IPv4 (preferIPv6Addresses defaults to false)
+                new ListenTest("localhost", ipv6Address)
                         .run(TestResult.AttachFailed);
                 // - listen on IPv6
                 new ListenTest("localhost", ipv6Address)
-                        .preferIPv6Addresses(true)
+                        .preferIPv6Addresses("true")
                         .run(TestResult.Success);
+                new ListenTest("localhost", ipv6Address)
+                        .preferIPv6Addresses("system")
+                        .run(systemPrefersIPv6 ? TestResult.Success : TestResult.AttachFailed);
             }
         } else {
             // IPv6-only system - expected to fail on IPv4 address
             new ListenTest("localhost", ipv6Address)
                     .preferIPv4Stack(true)
                     .run(TestResult.ListenFailed);
+            new ListenTest("localhost", ipv6Address)
+                    .preferIPv4Stack(true)
+                    .preferIPv6Addresses("system")
+                    .run(TestResult.ListenFailed);
+            new ListenTest("localhost", ipv6Address)
+                    .preferIPv4Stack(true)
+                    .preferIPv6Addresses("true")
+                    .run(TestResult.ListenFailed);
+            new ListenTest("localhost", ipv6Address)
+                    .run(TestResult.ListenFailed);
+            new ListenTest("localhost", ipv6Address)
+                    .preferIPv6Addresses("system")
+                    .run(systemPrefersIPv6 ? TestResult.Success : TestResult.AttachFailed);
```

Comment:
this is IPv6-only system, so systemPrefersIPv6 cannot be false

Reply:
Thanks, I removed the conditional and added an assertion that systemPrefersIPv6 if we get here